html/semantics/forms/form-submission-0/resources/file-submission.py

```diff
@@ -1,2 +1,9 @@
+from six import PY3
+
+from wptserve.utils import isomorphic_decode
+
 def main(request, response):
-    return ([("Content-Type", "text/html")], "<script>parent.postMessage(\"" + str(request.POST.first("testinput")) + "\", '*');</script>")
+    testinput = request.POST.first(b"testinput")
+    if PY3:
+        testinput.value = isomorphic_decode(testinput.value)
+    return ([(b"Content-Type", b"text/html")], u"<script>parent.postMessage(\"" + str(testinput) + u"\", '*');</script>")
```

Comment:
```suggestion
    return ([(b"Content-Type", b"text/html")], b"<script>parent.postMessage(\"" + request.POST.first("testinput") + b"\", '*');</script>")
```

This should remove the need for the `PY3` check and `isomorphic_decode`, by just sending the same bytes back. Or is there something more than meet the eye here?

Reply:
Agreed.

Reply:
_request.POST.first(b"testinput")_ returns _FieldStorage_ type. This will throw "_TypeError: can't concat FieldStorage to bytes_". 

_request.POST.first(b"testinput").value_ returns bytes in Python 3. That's why I thought to call isomorphic_decode() to convert it to a string before the _str(testinput)_ call in the original code.

Do please let me know if I have misunderstood anything here though.

Reply:
Oh this is a file!

Looking at where this is used [1], I think we actually have to call `str()` on the object instead of on its `value` property as the test is asserting the string representation of the object.

https://source.chromium.org/chromium/chromium/src/+/master:third_party/blink/web_tests/external/wpt/html/semantics/forms/form-submission-0/submit-file.sub.html;l=21;drc=77578ccb4082ae20a9326d9e673225f1189ebb63

Reply:
In python 3, the value property returns as a bytes type. I thought maybe we convert this property value to string before we call the object?

Reply:
Are you saying the string representations of this object in Python 2 and 3 are different? Could you paste the results of `str(request.POST.first(b"testinput"))` in Python 2 and 3 respectively?

Reply:
Yes. In Python 3 the printout is `FieldStorage('testinput', 'name', b'foobar')` while in Python 2 it is  `FieldStorage('testinput', 'name', 'foobar')`

Reply:
Ahh I see. This is rather annoying. Let's at least add a comment.